### ISSUE TITLE
Having an unsaved settings.json or an error there makes a very bad accessibility experience

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#147207]
I am distilling this from a conversation I had on the program-l mailing list for blind developers.

1. Have some error or some unsaved changes in your settings.json
2. Start VS Code, we ask the user if they are using a Screen Reader and if we should turn on accessibility
3. User says YES -> we can not write setting to the settings.json because it is dirty / has error
4. User can not fix settings.json because VS Code is not accessible 🐛 

Under 3 can we just auto save the settings.json before writing to it instead of asking?
Not sure if there is an error how to improve this flow. But this might come under a general problem of not easy to change settings if there is an error in settings.json

**Comment by sandy081** [comment]
I am wondering how did a user get into this state at point 1? Just want to make sure if this is realistic enough.

**Comment by isidorn** [comment]
Sorry but I do not know, this happened some time ago :)
Does settings sync, sync settings.json if it has errors? 

But the underlying problem is - we need to be able to write into settings even if there is an error. 
Feel free to close this as out of scope and if more users encounter it we can revisit.

**Comment by VSCodeTriageBot** [comment]
This issue has been closed automatically because it needs more information and has not had recent activity. See also our [issue reporting](https://aka.ms/vscodeissuereporting) guidelines.

Happy Coding!
